apps/pkcs12.c

```diff
@@ -878,15 +879,45 @@ int cert_load(BIO *in, STACK_OF(X509) *sk)
     return ret;
 }
 
+/* Generalised x509 attribute value print */
+
+void print_attr_type(BIO *out, const ASN1_TYPE *av)
```

Comment:
I think a name like `print_attribute` would be more suitable than `print_attr_type`. (It's a typed union, but the function does not print the attribute type, it prints the attribute (value)).

Reply:
Hmmm... but the argument _is_ an `ASN1_TYPE *`. The ASN1 naming conventions are sometimes really confusing. So never mind, just forget my last comment and leave the name as it is.

Reply:
_(I still find `print_attribute()` more intuitive, but I leave it up to you whether you want to change it or not.)_

Reply:
I agree it's confusing. An ASN1_OBJECT represents the type and the ASN1_TYPE is actually the value! Adding a commit to change the function name.